### PR TITLE
OIL-98: improve and test handling of locales (default configuration, fetching etc.)

### PR DESCRIPTION
**Pull request by ltparis2018**

### CODE REVIEW

File: test/specs/core/core_config.spec.js [hunk 1]
```diff
@@ -1,13 +1,98 @@
...
+  resetConfiguration,
+  getIabVendorListUrl,
+  getIabVendorBlacklist,
+  getIabVendorWhitelist,
+  getPoiGroupName,
+  getCookieExpireInDays,
+  getLocaleUrl,
+  getCustomPurposes,
+  getAdvancedSettingsPurposesDefault,
+  getDefaultToOptin } from '../../../src/scripts/core/core_config';
```
Comment (subsend) [inline test/specs/core/core_config.spec.js]: doppelte import?

File: test/specs/core/core_config.spec.js [hunk 2]
```diff
@@ -1,13 +1,98 @@
...
+import * as CoreConfig from '../../../src/scripts/core/core_config';
+import { loadFixture } from '../../utils.js';
```
Comment (subsend) [inline test/specs/core/core_config.spec.js]: wir müssen uns einigen entweder import mit *.js oder ohne :) beide nicht cool

File: README.md [hunk 3]
```diff
@@ -182,9 +183,11 @@ The labels are configured in you configuration. There are three options. The "lo
 <script>
 (function () {
     if (!window.AS_OIL) {
-      window.AS_OIL = {};
+      window.AS_OIL = {
+        CONFIG: {}
+      };
     }
-    window.AS_OIL.locale = {
+    window.AS_OIL.CONFIG.locale = {
```
Comment (CanardSauvage) [inline README.md]: Makes sense after some thoughts.

File: src/scripts/core/core_log.js [hunk 4]
```diff
@@ -24,6 +24,16 @@ export function logInfo() {
   }
 }
 
+/**
+ * LogWarn on none-production environments.
+ * if console.warn is not defined, fall back to console.log, ignore completely on weird cases
+ */
+export function logWarn() {
```
Comment (CanardSauvage) [inline src/scripts/core/core_log.js]: IMHO error or info should be enough. Log it as an error and it's fine.

File: src/scripts/userview/locale/userview_default_locale.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "localeId": "enEN_00",
+  "version": 0,
+  "texts": {
+    "label_intro_heading": "We use cookies and other technologies",
+    "label_intro_start": "label_intro_start",
```
Comment (CanardSauvage) [inline src/scripts/userview/locale/userview_default_locale.json]: We should add more documentation about the labels, like
label_intro_start + privacy_page_url + label_intro_end vs. label_intro --> label_intro has higher priority

**Comment by CanardSauvage** [comment]
* I would remove the logWarn method
* We should add way more documentation about the labels!

**Pull request merged**